taxi/views.py

```diff
@@ -87,3 +94,40 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
 class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
     queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+
+
+class DriverCreateView(LoginRequiredMixin, generic.CreateView):
+    model = Driver
+    form_class = DriverCreationForm
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class DriverDeleteView(LoginRequiredMixin, generic.DeleteView):
+    model = Driver
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class DriverUpdateView(LoginRequiredMixin, generic.UpdateView):
+    model = Driver
+    form_class = DriverLicenseUpdateForm
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class AssignCarView(LoginRequiredMixin, generic.View):
+    @staticmethod
+    def get(request, car_id):
+        current_driver = Driver.objects.get(id=request.user.id)
+        car = Car.objects.get(id=car_id)
+        car.drivers.add(current_driver)
+
+        return redirect("taxi:car-detail", car.id)
+
+
+class DeleteCarView(LoginRequiredMixin, generic.View):
+    @staticmethod
+    def get(request, car_id):
```

Comment:
Use POST method. GET is used to retrieve a resource, not to change it

You can even use single AssingnDeleteCarView